kedro/contrib/idempotent/idempotent_state_storage.py

```diff
@@ -3,39 +3,60 @@
 from uuid import uuid4
 from typing import List, Any
 
-from kedro.io.data_catalog import MemoryDataSet
+from kedro.io.data_catalog import MemoryDataSet, DataCatalog
+
+IDEMPOTENT_STATE_STORAGE_CATALOG_NAME = 'idempotent_state_storage'
 
 
 class IdempotentStateStorage:
-    def __init__(self, run_id_state=None, input_state=None, nodes_have_been_run=None):
-        if run_id_state is None:
-            run_id_state = {
-                # "ds1": "qwer-qwer-wqer",
-                # "ds2": "adsf-asdf-adsf"
-            }
-        if input_state is None:
-            input_state = {
-                # "node1": {},
-                # "node2": {
-                #     "ds1": "qwer-qwer-qwer"
-            }
-        if nodes_have_been_run is None:
-            nodes_have_been_run = {
-                # "node1": True
-            }
+    def __init__(self, catalog: DataCatalog, state_catalog_entry: str = None):
+        self.catalog = catalog
+        if state_catalog_entry is None:
+            state_catalog_entry = IDEMPOTENT_STATE_STORAGE_CATALOG_NAME
+        self.state_catalog_entry = state_catalog_entry
+
+        try:
+            state_data = catalog.load(IDEMPOTENT_STATE_STORAGE_CATALOG_NAME)
+        except Exception as e:
+            print(
```

Comment:
Don't use a print statement here, just raise a new exception. Create an `IdempotentStateStorageException` and call the exception with it.

`IdempotentStateStorageException("Failed to load ...")`